bazel/test.bzl

```diff
@@ -99,15 +123,19 @@ def _redpanda_cc_test(
         data = data,
     )
 
-def _redpanda_cc_unit_test(cpu, **kwargs):
+def _redpanda_cc_unit_test(cpu, memory, **kwargs):
     extra_args = [
         "--unsafe-bypass-fsync 1",
         "--default-log-level=trace",
         "--logger-log-level='io=debug'",
         "--logger-log-level='exception=debug'",
     ]
+
+    # TODO(rockwood): Should we default to a
+    # more reasonable unit test resource like a
+    # single core and 256MiB of memory?
```

Comment:
memory: we could probably move it progressively lower until we found a workable minimum. i'm not sure what we'll find when we get fixture tests added.

cpu: using a single core has bitten us in the past when code had bugs that only appeared with multiple cores, and worked fine with a single core. 3 might be nice. i dunno. we could also create finer grained categories of tests than what we have now.

Reply:
Simplified the comment to just re-consider these defaults. I don't think it's super important right now, but I think maybe we want finer gained, because most of the unit tests I write explicitly only run a single shard, so allocating more is useless. Maybe we just shouldn't have defaults and force every test to be annotated properly. IDK.